staging/cse/windows/kubeletfunc.ps1

```diff
@@ -209,6 +209,70 @@ function Get-KubePackage {
     Remove-Item $zipfile
 }
 
+function Remove-KubeletNodeLabel {
+    Param(
+        [Parameter(Mandatory=$true)][string]
+        $KubeletNodeLabels,
+        [Parameter(Mandatory=$true)][string]
+        $Label
+    )
+
+    if ($KubeletNodeLabels -Match ",$Label") {
+        return $KubeletNodeLabels -replace ",$Label", ""
+    }
+
+    if ($KubeletNodeLabels -Match "$Label,") {
+        return $KubeletNodeLabels -replace "$Label,", ""
+    }
+
+    if ($KubeletNodeLabels -Match "$Label") {
+        return $KubeletNodeLabels -replace "$Label", ""
+    }
+
+    return $KubeletNodeLabels
+}
+
+function Get-TagValue {
+    Param(
+        [Parameter(Mandatory=$true)][string]
+        $TagName,
+        [Parameter(Mandatory=$true)][string]
+        $DefaultValue
+    )
+
+    $uri = "http://169.254.169.254/metadata/instance?api-version=2021-02-01"
+    $response = Retry-Command -Command "Invoke-RestMethod" -Args @{Uri=$uri; Method="Get"; ContentType="application/json"; Headers=@{"Metadata"="true"}} -Retries 3 -RetryDelaySeconds 5
+
+    $tag = $response.compute.tagsList | Where-Object { $_.name -eq $TagName }
+    if (!$tag) {
+        return $DefaultValue
+    }
+    return $tag.value
+}
+
+function Disable-KubeletServingCertificateRotationForTags {
+    Logs-To-Event -TaskName "AKS.WindowsCSE.DisableKubeletServingCertificateRotationForTags" -TaskMessage "Check whether to disable kubelet serving certificate rotation via nodepool tags."
+
+    Write-Log "Checking whether to disable kubelet serving certificate rotation for nodepool tags"
+
+    if (!($global:EnableKubeletServingCertificateRotation)) {
+        Write-Log "Kubelet serving certificate rotation is already disabled"
+        return
+    }
+
+    $tagName = "aks-disable-kubelet-serving-certificate-rotation"
+    $disabled = Get-TagValue -TagName $tagName -DefaultValue "false"
+    if (!($disabled -eq "true")) {
+        Write-Log "Nodepool tag `"$tagName`" is missing or not set to true, nothing to disable"
+        return
+    }
+
+    Write-Log "Kubelet serving certificate rotation is disabled by nodepool tags, will reconfigure kubelet flags and node labels"
+
+    $global:KubeletConfigArgs = $global:KubeletConfigArgs -replace "--rotate-server-certificates=true", "--rotate-server-certificates=false"

```

Comment:
Should we set it to false whem current args do not contain `--rotate-server-certificates`?

Reply:
I was thinking we shouldn't set it to anything if the flag isn't already there (which in this case is fine since it defaults to false anyways)

that way we won't accidentally give this flag to a kubelet version that doesn't support it